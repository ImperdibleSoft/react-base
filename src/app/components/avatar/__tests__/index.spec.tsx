import * as React from 'react';
import { shallow } from 'enzyme';

import Avatar from '..';

describe('Avatar', () => {
  describe('with no valid picture', () => {
    it('should render correctly a company placeholder', () => {
      const component = shallow(<Avatar profile={{ picture: '', name: 'Company Inc.' }} />);
      expect(component).toMatchSnapshot();
    });

    it('should render correctly a user placeholder', () => {
      const component = shallow(<Avatar profile={{ picture: '', fullName: 'My full name' }} />);
      expect(component).toMatchSnapshot();
    });
  });

  describe('with valid picture', () => {
    it('should render correctly a company Avatar', () => {
      const component = shallow(<Avatar profile={{ picture: 'my-picture', name: 'Company Inc.' }} />);
      expect(component).toMatchSnapshot();
    });

    it('should render correctly a user Avatar', () => {
      const component = shallow(<Avatar profile={{ picture: 'my-picture', fullName: 'My full name' }} />);
      expect(component).toMatchSnapshot();
    });
  });
});
