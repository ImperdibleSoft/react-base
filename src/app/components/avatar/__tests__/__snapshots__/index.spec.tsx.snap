// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar with no valid picture should render correctly a company placeholder 1`] = `
<Avatar
  avatarBorder={7.5}
  avatarFontSize={36.00000576000092}
  avatarIconSize={42.000100800241924}
  avatarSize={75}
  isEmpty={true}
  isExpandable={false}
>
  <i
    className="fa fa-globe"
  />
</Avatar>
`;

exports[`Avatar with no valid picture should render correctly a user placeholder 1`] = `
<Avatar
  avatarBorder={7.5}
  avatarFontSize={36.00000576000092}
  avatarIconSize={42.000100800241924}
  avatarSize={75}
  isEmpty={true}
  isExpandable={false}
>
  <i
    className="fa fa-user"
  />
</Avatar>
`;

exports[`Avatar with valid picture should render correctly a company Avatar 1`] = `
<Avatar
  avatarBorder={7.5}
  avatarFontSize={36.00000576000092}
  avatarIconSize={42.000100800241924}
  avatarSize={75}
  isEmpty={false}
  isExpandable={false}
>
  <AvatarPicture
    src="my-picture"
  />
</Avatar>
`;

exports[`Avatar with valid picture should render correctly a user Avatar 1`] = `
<Avatar
  avatarBorder={7.5}
  avatarFontSize={36.00000576000092}
  avatarIconSize={42.000100800241924}
  avatarSize={75}
  isEmpty={false}
  isExpandable={false}
>
  <AvatarPicture
    src="my-picture"
  />
</Avatar>
`;
